dotcom-rendering/src/web/lib/braze/buildBrazeMessaging.ts

```diff
@@ -89,18 +105,22 @@ export const buildBrazeMessages = async (
 		const errorHandler = (error: Error, desc: string) => {
 			window.guardian.modules.sentry.reportError(error, desc);
 		};
-		const brazeMessages = new BrazeMessages(
-			appboy,
-			LocalMessageCache,
-			errorHandler,
-		);
 
 		setHasCurrentBrazeUser();
 		appboy.changeUser(dependenciesResult.data.brazeUuid as string);
 		appboy.openSession();
 
-		return brazeMessages;
+		const brazeCards = new BrazeCards(appboy, errorHandler);
+		const brazeMessages = new BrazeMessages(
+			appboy,
+			LocalMessageCache,
+			errorHandler,
+		);
+		return { brazeMessages, brazeCards };
 	} catch {
-		return new NullBrazeMessages();
+		return {
+			brazeMessages: new NullBrazeMessages(),
+			brazeCards: new NullBrazeCards(),
+		};
```

Comment:
I find early returns easy to reason about, and you can achieve a similar thing with try-catch blocks.

Instead of returning this in three different instances, including the two failure states–no local storage, or the depedencies check are unsucessful–you could throw on failure and find yourself in the catch block. You would have to wrap the whole function body inside the try block.

Reply:
Ah that's interesting. I've always slightly shied away from using exceptions for flow control as I think it _can_ make it harder to follow the flow of the code. Is it a pattern used elsewhere in DCR? I do agree that the multiple points where we return the same thing is a bit sub-optimal, I'll see if I can improve that.

Reply:
I don’t think it’s a pattern I would recommend, but in some cases it may make sense? Again, we’ve got TypeScript making sure that the return types match, so it’s not really worth improving on much further here, as it’s only duplicated twice and not worth creating an abstraction over.